src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -111,8 +113,16 @@ class DefinitionV11(DefinitionV10):
 
 
 class DefinitionV20(_ProjectDefinitionBase):
-    entities: Dict[str, Annotated[Entity, Field(discriminator="type")]] = Field(
-        title="Entity definitions."
+    entities: Dict[str, AnnotatedEntity] = Field(title="Entity definitions.")
+
+    defaults: Optional[DefaultsField] = Field(
+        title="Default key/value entity values that are merged recursively for each entity.",
+        default=None,
+    )
+
+    env: Optional[Dict[str, Union[str, int, bool]]] = Field(
+        title="Default environment specification for this project.",
+        default=None,
```

Comment:
Is this really being added in this PR? How does it releat to Snowpark?

Reply:
It was moved from line 165. It seems a bit more consistent to have attributes first, then methods - not attribute, method, attribute, method etc.

Reply:
Let's limit the changed to snowpark only. It's easier to revert or review when changes has well defined scope.

Reply:
Done